### DISCUSSION TITLE
Possibility to pause/stop the SPI interface?

### DISCUSSION THREAD

**Discussion by aster1cs** [Bodmer/TFT_eSPI#2464] [New processor/display support or capability request] [answered]
I'm having an Arduino project (esp32 development board) that uses the great TFT_eSPI library for controlling an ILI9488 via SPI, which works nicely.
However, in that project I am using the same pins used for my SPI bus (I'm running short on GPIOs) for alternative purposes (controlling some other ICs via discrete high/low states of the GPIOs).
The display updates and the other tasks of the GPIOs are alternating, so there is no time overlap between SPI communication and measurements.
However, in order to properly stop/pause the SPI, I needed to add a method to the TFT_eSPI.cpp:

```
void TFT_eSPI::closeSpi()
{
	spi.end();
	_booted = true;
}
```

If I do not end the SPI connection, it will interfere with my alternative usage of the GPIOs. (Obviously I'd need to call myDisplay.init() again once I want to update the TFT.)

So the actual question: Does the library already offer some functionality to pause/end an SPI connection which I didn't recognise or is this functionality really missing and adding something like closeSpi() the way to go?

**Comment by rradomir** [comment, marked as answer]
Did you tried to tie other connection by tft.getSPIinstance() ?
For example I have sd card reader and this work perfectly: `SD.begin(SD_CS, tft.getSPIinstance())`

**Reply by aster1cs** [reply]
Hello rradomir,

thanks for the hint. However, the other connection in my case is not an SPI connection. Instead, the GPIO pins are used to address an external multiplexer-IC which is disabled during the TFT updates.

Update:
I found that calling .end() and .begin(...) on the returned reference of getSPIinstance() seems to work for my purpose. Thanks for the idea! This way I don't need any modification of the library.
